lib/segment/src/spaces/metric_uint/sse2/cosine.rs

```diff
@@ -7,8 +7,6 @@ use crate::spaces::simple_sse::hsum128_ps_sse;
 #[allow(clippy::missing_safety_doc)]
 pub unsafe fn sse_cosine_similarity_bytes(v1: &[u8], v2: &[u8]) -> f32 {
     debug_assert!(v1.len() == v2.len());
-    debug_assert!(is_x86_feature_detected!("sse"));
-    debug_assert!(is_x86_feature_detected!("sse2"));
```

Comment:
```
error: this assertion has a constant value
  --> lib/segment/src/spaces/metric_uint/sse2/manhattan.rs:11:5
   |
11 |     debug_assert!(is_x86_feature_detected!("sse2"));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider moving this into a const block: `const { assert!(..) }`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/beta/index.html#assertions_on_constants
```

we do not have those debug assertions in other metrics implementations so I just removed those

Reply:
I argue this is a bad lint, and it shouldn't consider this to be a constant.

But I have no objection to removing it.